pkg/cri/store/container/container.go

```diff
@@ -188,7 +192,12 @@ func (s *Store) UpdateContainerStats(id string, newContainerStats *stats.Contain
 	}
 
 	c := s.containers[id]
-	c.Stats = newContainerStats
+	if c.Stats != nil {
```

Comment:
why do we need to check if `c.Stats` is nil here? Shouldn't it be initialized previously?

Reply:
I believe this checks for the case where old stats is nil [here](https://sourcegraph.com/github.com/containerd/containerd@bb45abaa2cbc687160a918030af1b9b9f99f7ef4/-/blob/pkg/cri/server/container_stats_list.go?L99:5)

Reply:
Also get a memory panic if I take it out

Reply:
Hmm, where do we actually assign `c.Stats`? Is it ever assigned?

As I see we are only returning `*containerstorestats.ContainerStats` in `generatedContainerMetrics`, but we are not actually storing the stats on the container object? Is that right? If that's the case, then c.Stats will always be nil...

Reply:
Yep, good point -- actual caching of the metrics should be fixed now.